### PR TITLE
 adding my tests cases for R2-Register, R7, and R8

### PR DESCRIPTION
**Pull request by Wiederickj**
# Pull Request Template

## Description

Added a new file in qa327_test for testing the frontend register function (R2). Also added R7 and R8 to run test cases for logout and the 404 error page.

Added a failure report. Made some changes to the source code. Like adding error 404 pages

Fixes # (issue)

## Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update

## How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

Reran code after changes. Observed web app to determine if changes were made.

- [x] Test A
- [x] Test B

**Test Configuration**:
* Firmware version:
* Hardware:
* Toolchain:
* SDK:

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules
- [x] I have checked my code and corrected any misspellings

### CODE REVIEW

File: qa327_test/frontend/test_r8.py [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+@pytest.mark.usefixtures('server')
+class R8TestCase(BaseCase):
+    def test_error_page_register(self):
+        self.open(base_url + '/Register')
+        self.assert_element("#welcome-header")
+        self.assert_text("404 - Page Not Found!", "#welcome-header")
+    
+    
+    def test_error_page_wallet(self):
+        self.open(base_url + '/Wallet')
```
Comment (HamzaSherik7) [inline qa327_test/frontend/test_r8.py:17]: is /wallet a new page?

**Review by HamzaSherik7 — APPROVED** [review]
Good work!

File: qa327_test/frontend/test_r8.py (hunk 1 above)
Comment (Wiederickj) [reply qa327_test/frontend/test_r8.py:17]: yes just for testing purposes

**Review by Gilbert-Chien — COMMENTED** [review]
Looks good. Great work!

**Review by Gilbert-Chien — APPROVED** [review]

**Pull request merged**